cdap-app-fabric/src/main/java/co/cask/cdap/metadata/LineageHTTPHandler.java

```diff
@@ -101,11 +103,22 @@ public void datasetFields(HttpRequest request, HttpResponder responder,
                             @PathParam("dataset-id") String datasetId,
                             @QueryParam("start") String startStr,
                             @QueryParam("end") String endStr,
-                            @QueryParam("prefix") String prefix) throws BadRequestException {
+                            @QueryParam("prefix") String prefix,
+                            @QueryParam("includeDatasetSchema") boolean includeDatasetSchema)
```

Comment:
this is a bit long and a little redundant, as 'datasets' is in the path already. It also ties the response format to functionality. For example, there is no way to to exclude fields without lineage but in the new format.

How was this done for the metadata APIs? Is the default response format the new format or the old one? If the default is the old, how/when do we plan on moving to the new?

Reply:
In Metadata we changed the response to return MetadataEntity rather than EntityId. The default behavior is to return EntityId which will be changed once the UI is changed to understand MetadataEntity.

Reply:
maybe we can name this includeCurrent, as the behavior change is that all current fields will be included in the output.